### PR TITLE
Add basic WAMP client

### PR DESCRIPTION
**Pull request by muzzammilshahid**

### CODE REVIEW

File: lib/src/session.dart [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+  final BaseSession _baseSession;
+  late WAMPSession _wampSession;
+
+  final SessionScopeIDGenerator _idGen = SessionScopeIDGenerator();
+
+  int get _nextID => _idGen.next();
+
+  final Map<int, Completer<Result>> _callRequests = {};
+  final Map<int, RegisterRequest> _registerRequests = {};
+  final Map<int, Yield Function(Invocation)> _registrations = {};
```
Comment (om26er) [inline lib/src/session.dart:31]: As far as user api is concerned this should have a new class Result AND Invocation defined in types. So instead of Yield we will use Result. NOTE Result wont be a wamp message type but rather a normal class. See wamp.py

**Review by om26er — CHANGES_REQUESTED** [review]
TL;DR we need new types Result, Invocation and Event. They wont include request id etc

**Review by om26er — APPROVED** [review]
